Rewrite docs on organizations and projects, with per-project permissions

Twixes opened the pull request:
## Changes

Completely rewritten docs on organizations and projects to make them much more comprehensive.

Also, importantly, included section "Per-project permissions" – this outlines the solution to https://github.com/PostHog/posthog/issues/4263 that I'm currently implementing. Feel free to suggest any change to part, as all details can still be changed.

## Checklist
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Words are spelled using American english
- [x] I have checked out our [styleguide](https://github.com/PostHog/posthog.com/blob/master/STYLEGUIDE.md)

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -65,23 +65,23 @@ This makes for the smoothest collaboration. However, if you require access to da
+3. **Owner** – unique top level, like Administrator plus:
+  - can delete the organization
+  - cannot leave the organization
+  - can pass ownership to someone else, ridding themselves of it
```
Ridding sounds like it's undesirable

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -65,23 +65,23 @@ This makes for the smoothest collaboration. However, if you require access to da
+  - can always create and delete projects
+  - cam manage access levels of other members
+  - can add and remove other members from projects (if per-project access mode enabled)
+  - can remove other members from the organization
```
Including other admins but not the owner

paolodamico reviewed (changes requested):
Thanks @Twixes, this makes it a lot clearer! Minor comments inline, in addition project-based permissions should be an `ee` feature (see https://github.com/PostHog/company-internal/issues/417)

Twixes commented:
Good points, addressed @paolodamico

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -91,3 +94,5 @@ With **Per-project access** enabled, nothing changes for organization members at
 
 In this mode you can also grant members project-specific access level that is _above_ their organization-wide one. In such case they will have privileges of the higher level, but only for that specific project.
+
+> **Note:** Per-project access belongs to our premium team-oriented offering. To use this feature, [set up PostHog Cloud billing](https://posthog.com/pricing) or [contact us](https://share.hsforms.com/1-IVCY9gNRvaZBajMt_UPIg4559u) for a self-hosted license.
```
I think it's better to take the user to the pricing page either way, in case we letter change the form this doesn't get outdated.

paolodamico replied:
One additional thing. We don't specify how to enable/disable per-project access.

Twixes replied:
Make sense re: first point. Though that form link is still in a ton of places on the website.
Re: second point, we did specify a few paragraphs above, but I made it stand out more.

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+showTitle: true
+---
+
+For a user experience that scales for all your needs, PostHog implements the concepts of organizations and projects.
```
> or a user experience that scales for all your needs

This doesn't tell me why I want these concepts. Can we provide a better overview of why Organizations and Projects are valuable?

Twixes replied:
Very vague, agreed. I kept it very concise, but explained better now

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+As a brand new PostHog user you'll always find yourself in a single organization upon account creation.
+If you've joined from an invite, that will be the organization you've been invited to.
+Otherwise you'll become the Owner of a new organization created just for you, based on the company name you provided when creating the account.
```
```suggestion
Otherwise, you'll become the Owner of a new organization created just for you, based on the company name you provided when creating the account.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+If you've joined from an invite, that will be the organization you've been invited to.
+Otherwise you'll become the Owner of a new organization created just for you, based on the company name you provided when creating the account.
+
+To switch between organizations, to go the current organization's settings, or to create new projects, use the account dropdown on the right of the top bar.
```
```suggestion
To switch between organizations, navigate to the current organization's settings, or to create new projects, use the account dropdown on the right of the top bar.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+By default every time a member joins an organization in an [email-enabled PostHog instance](/docs/self-host/configure/email) (including PostHog Cloud), all existing members of the organization get notified about this with an email. This increases security by making it explicit who gains access to your product data.
+
+If you'd rather not get such notifications however, you can simply disable setting "Email all current members when a new member joins" in section "Notification Preferences" of organization settings.
```
```suggestion
If you'd rather not get such notifications, you can disable setting "Email all current members when a new member joins" in the section "Notification Preferences" of organization settings.
```

Twixes replied:
I believe it should be either "in the XYZ section" or "in section XYZ".

leggetter replied:
👍  Updated suggestion

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+Any organization member can create organization invites. Such an invite is valid **for 3 days** after creation and **only for the specified email**.
+
+If there's no account associated with that email, the invited person will have to create an account. Otherwise they'll be able to join with their existing account.
```
```suggestion
If there's no account associated with that email, the invited person will have to create an account. Otherwise, they'll be able to join with their existing account.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+Every project has its own distinct write-only token, which you can use to initialize your [integration of choice](/docs/integrate), as well as to connect to our [API](/docs/api). You can always regenerate this token, but keep in mind that the old one will be immediately revoked.
+
+Every new organization (including the one created for you on account creation) comes with a fresh project named "Default Project". You can then rename or delete it as you see fit.
```
```suggestion
Every new organization (including the one created for you on account creation) comes with a fresh project named "Default Project". You can rename or delete it as you see fit.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+Every new organization (including the one created for you on account creation) comes with a fresh project named "Default Project". You can then rename or delete it as you see fit.
+
+To switch between projects, to go their settings, or to create new projects, use the project switcher in the middle of the top bar. You can also quickly go to the current project's settings from the sidebar.
```
```suggestion
To switch between projects, navigate to project settings, or to create new projects, use the project switcher in the middle of the top bar. You can also quickly go to the current project's settings from the sidebar.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+To switch between organizations, to go the current organization's settings, or to create new projects, use the account dropdown on the right of the top bar.
+
+> **Note:** As a PostHog Cloud user, you can create, manage, and join organizations without limits forever. For self-hosted PostHog, however, multiple organizations belong to our premium team-oriented offering. To use this feature, [contact us for a self-hosted license](https://posthog.com/pricing).
```
```suggestion
> **Note:** As a PostHog Cloud user, you can create, manage, and join organizations without limits forever. For self-hosted PostHog, however, multiple organizations belong to our premium team-oriented offering. To use this feature, [contact us for a self-hosted license](/pricing).
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+To switch between projects, to go their settings, or to create new projects, use the project switcher in the middle of the top bar. You can also quickly go to the current project's settings from the sidebar.
+
+> **Note:** Multiple projects within an organization belong to our premium team-oriented offering. To use this feature, [set up PostHog Cloud billing](https://app.posthog.com/organization/billing) or [contact us for a self-hosted license](https://posthog.com/pricing).
```
```suggestion
> **Note:** Multiple projects within an organization belong to our premium team-oriented offering. To use this feature, [set up PostHog Cloud billing](https://app.posthog.com/organization/billing) or [contact us for a self-hosted license](/pricing).
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+- A project per environment – you may want to test out analytics instrumentation in development or staging environments, while keeping that test data separate from production.
+- A project per product – if you have several unlinked products, it makes sense to reflect that separation in PostHog too.
+- A project per multiple products or the whole company – if your offerring is interlinked or you want to track the user journey holistically (e.g. how many company blog readers convert to paid product users), you can use a single project for all that.
```
```suggestion
- A project per multiple products or the whole company – if your offering is interlinked or you want to track the user journey holistically (e.g. how many company blog readers convert to paid product users), you can use a single project for all digital assets.
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+2. **Administrator** – elevated level, like Member plus:
+  - can always create and delete projects
+  - can change project and organization settings
+  - cam manage access levels of other members (except the Owner)
```
```suggestion
  - can manage access levels of other members (except the Owner)
```

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+By default all organization members have access to all its projects, at their organization-wide access level.
+This makes for the smoothest collaboration. However, if you require access to data to be more granular, see section [Per-project access](#per-project-access) below.
+
+### Access levels
```
How many of each level can there be?

Twixes replied:
Clarified

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,102 @@
+
+In this mode you can also grant members project-specific access level that is _above_ their organization-wide one. In such case they will have privileges of the higher level, but only for that specific project.
+
+> **Note:** Per-project access belongs to our premium team-oriented offering. To use this feature, [set up PostHog Cloud billing](https://app.posthog.com/organization/billing) or [contact us for a self-hosted license](https://posthog.com/pricing).
```
```suggestion
> **Note:** Per-project access belongs to our premium team-oriented offering. To use this feature, [set up PostHog Cloud billing](https://app.posthog.com/organization/billing) or [contact us for a self-hosted license](/pricing).
```

leggetter reviewed (changes requested):
A few small but worthwhile changes requested.

Generally really well explained.

Two main comments:

- The intro paragraph needs to do more to explain the value of these concepts and organisational constructs
- We should probably capitalise Organization and Project when being used as a noun

Twixes commented:
Good points @leggetter, I'm not sure about capitalizing these words here though.
The idea has some merit to me, as they're sort of distinct concepts, but I think this should be clearly explained in the style guide if we want to do this, because there's more concepts like this, and there are some edge cases. For instance, should it be "Organization invite" or "Organization Invite". Also, since there's access level "Member" I use lower-case "member" as "any organization member" and "Member" as "Member-level organization member", which would be lost if we capitalized that noun too.
So I'd rather do that in another PR.

leggetter commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,103 @@
+
+### Access levels
+
+There are 3 access levels in PostHog, each with more privileges than the one below it. There can be any number of members at each level, except Owner, of which there can only be one in the organization.
```
Nit: 3 vs three?

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+## Organizations
+
+An organization is the largest building block of PostHog's structure: it's made up of projects and has PostHog users as members. 
```
```suggestion
An organization is the highest abstraction level within a PostHog instance. An organization commonly represents a real-world company or other type of grouping that is completely isolated. An organization can have multiple projects, which can be used to logically separate data (e.g. mobile/web apps, teams within a company, etc.). Team members belong to an organization.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+An organization is the largest building block of PostHog's structure: it's made up of projects and has PostHog users as members. 
+Each organization member has an organization-wide [access level](#access-levels) dictating their ability to perform certain actions.
+
+Most commonly a PostHog organization represents a real-world company, but this isn't a requirement and you are free to gather people any way you see fit.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+An organization is the largest building block of PostHog's structure: it's made up of projects and has PostHog users as members. 
+Each organization member has an organization-wide [access level](#access-levels) dictating their ability to perform certain actions.
+
```
Maybe worth asking @lottiecoxon for a simple art here that represents the hierarchy of PostHog instance -> org -> project (and members)?

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+## Organizations
+
+An organization is the largest building block of PostHog's structure: it's made up of projects and has PostHog users as members. 
+Each organization member has an organization-wide [access level](#access-levels) dictating their ability to perform certain actions.
```
```suggestion
Permissions can be handled both at the organization or project level (see below), and a user's ability to take an action is determined based on both.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+Most commonly a PostHog organization represents a real-world company, but this isn't a requirement and you are free to gather people any way you see fit.
+
+As a brand new PostHog user you'll always find yourself in a single organization upon account creation.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+Most commonly a PostHog organization represents a real-world company, but this isn't a requirement and you are free to gather people any way you see fit.
+
+As a brand new PostHog user you'll always find yourself in a single organization upon account creation.
+If you've joined from an invite, that will be the organization you've been invited to.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+As a brand new PostHog user you'll always find yourself in a single organization upon account creation.
+If you've joined from an invite, that will be the organization you've been invited to.
+Otherwise, you'll become the Owner of a new organization created just for you, based on the company name you provided when creating the account.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+If you've joined from an invite, that will be the organization you've been invited to.
+Otherwise, you'll become the Owner of a new organization created just for you, based on the company name you provided when creating the account.
+
+To switch between organizations, navigate to the current organization's settings, or to create new projects, use the account dropdown on the right of the top bar.
```
```suggestion
You can **have access to multiple organizations with a single account** (note that all data is still segmented between organizations). To switch between organizations click on the account dropdown on the top right corner.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+To switch between organizations, navigate to the current organization's settings, or to create new projects, use the account dropdown on the right of the top bar.
+
+> **Note:** As a PostHog Cloud user, you can create, manage, and join organizations without limits forever. For self-hosted PostHog, however, multiple organizations belong to our premium team-oriented offering. To use this feature, [contact us for a self-hosted license](/pricing).
```
```suggestion
> **Note:** As a PostHog Cloud user, you can create, manage, and join multiple organizations without limits. For self-hosted PostHog, however, multiple organizations belong to our [premium](/pricing) offering.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+If there's no account associated with that email, the invited person will have to create an account. Otherwise, they'll be able to join with their existing account.
+
+Newly-joined users get the basic Member access level.
+
```
Maybe clarify when we send the invite email automatically and when we only generate the invite link

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+## Permissions
+
+By default all organization members have access to all its projects, at their organization-wide access level.
```
```suggestion
By default all organization members have access to all its projects.
```
& remove new line below, continue in a single paragraph

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+### Access levels
+
+There are three access levels in PostHog, each with more privileges than the one below it. There can be any number of members at each level, except Owner, of which there is always one in the organization.
```
```suggestion
There are three access levels in PostHog (Owner, Admin & Member). Please note that there can only be one Onwer per organization.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+See permissions at each level below:
+
+| | Member (base level) | Administrator (elevated level) | Owner (unique top level) |
```
```suggestion
| | Member (base level) | Administrator | Owner |
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+Access levels can be viewed and changed in the Members section of organization settings.
+
+In projects with [project-based permissioning](#project-based-permissioning), there's also a Members section in project settings, which pertains to project-specific memberships.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+### Project-based permissioning
+
+If you'd like to restrict access to data within the organization to only those who need it, you can use PostHog's project-based permissioning.
```
```suggestion
If you'd like to restrict access to data within the organization to only those who need it, you can use PostHog's project-based permissioning. Projects with this option enabled will become **secret by default and invite-only** (except for Administrators and the Owner).
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+If you'd like to restrict access to data within the organization to only those who need it, you can use PostHog's project-based permissioning.
+
+**Enable project-based permissioning** is a switch in project settings that can be turned on (or off) at any time by organization-wide Administrators and the Owner.
```
```suggestion
Any Administrator or Owner can make a project private or open at any time.
```

paolodamico replied:
I'm suggesting here from a user standpoint to refer to this setting as having "Private" or "Open" projects, instead of project-based permissioning.

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+In projects with [project-based permissioning](#project-based-permissioning), there's also a Members section in project settings, which pertains to project-specific memberships.
+
+### Project-based permissioning
```
```suggestion
### Private projects (project-based permissioning)
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+**Enable project-based permissioning** is a switch in project settings that can be turned on (or off) at any time by organization-wide Administrators and the Owner.
+
+With project-based permissioning only organization-wide Administrators and the Owner have implicit access to projects.
```
```suggestion
Private projects can **only be accessed by Administrators and Owners by default**. Regular members can be added to projects from the project settings page to gain access.
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+**Enable project-based permissioning** is a switch in project settings that can be turned on (or off) at any time by organization-wide Administrators and the Owner.
+
+With project-based permissioning only organization-wide Administrators and the Owner have implicit access to projects.
+Lower-level Members need to be added explicitly in project settings. Those that haven't been added cannot access the project at all, in any way.
```
```suggestion
```

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,96 @@
+
+With project-based permissioning only organization-wide Administrators and the Owner have implicit access to projects.
+Lower-level Members need to be added explicitly in project settings. Those that haven't been added cannot access the project at all, in any way.
+At the same time project-based permissioning allows you to grant members project-specific access _higher_ than their organization-wide level.
```
```suggestion
Regular organization members can also be made administrators of specific projects, and they will be able to manage users who have access to that specific project.
```

paolodamico reviewed (commented):
Made suggestions inline, but looking pretty solid!

Twixes commented:
Alright, used some suggestions but also changed a few. @paolodamico I would suggest with this number of suggestions to just create a PR on top of the one being edited in the future, as an article loses cohesiveness with multiple individual-line edits.

Added the PR to the Artwork board, not high priority, but it could be nice to have an illustration (referring to https://github.com/PostHog/posthog.com/pull/1948#discussion_r713603299).

paolodamico commented on contents/docs/user-guides/organizations-and-projects.md:
```diff
@@ -0,0 +1,90 @@
+
+### Private projects
+
+If you'd like to restrict access to data within the organization to only those who need it, you can make a project _private_. Projects with this setting enabled become **secret by default and invite-only** (except for Administrators and the Owner, who get access implicitly).
```
```suggestion
If you'd like to restrict access to data within the organization to only those who need it, you can make a project _private_. Projects with this setting enabled become **private by default and invite-only** (except for Administrators and the Owner, who get access implicitly).

I'm sorry, I know I made this suggestion but just realized secret implies that it's completely hidden from other members, but it's not secret (other team members can still see it exists and its name), it's private
```

Twixes replied:
Ah, I thought it was a conscious suggestion to make them secret in the product too, but very well, let's just keep them private as currently implemented.

paolodamico replied:
I think it's fine to keep it like this, we can get more feedback from users in the upcoming weeks as to whether there's a use case for secret projects, but my guess is that it won't.

paolodamico reviewed (approved):
Thanks @Twixes! Definitely, my apologies didn't realize I would be making many suggestions.

leggetter commented:
@Twixes - is this just about good to be merged?

Twixes commented:
It would be, but first we need to release https://github.com/PostHog/posthog/issues/4263 to the public. :) These docs are false until the feature is released.
